persistedsqlstats: TestSQLStatsFlush is flaky

```
No pass/skip/fail event found for test
=== RUN   TestSQLStatsFlush
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsFlush3569017073
    test_log_scope.go:81: use -show-logs to present logs inline
panic: test timed out after 5m24s
running tests:
```

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/13055735?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

Jira issue: CRDB-34371

--- a/pkg/sql/sqlstats/persistedsqlstats/flush_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/flush_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -66,6 +67,8 @@ var testQueries = []testCase{
 func TestSQLStatsFlush(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "test uses multi-node cluster; takes too long to run")
 
 	fakeTime := stubTime{
 		aggInterval: time.Hour,

